### ISSUE TITLE
Mob Respawn Time

### ISSUE THREAD

**Issue by Sinoxan** [classicdb/database#378]
The respawn time for regular mobs seems too fast.  I'm not sure what it should be but mobs in Elwynn Forest were respawning after only 120 seconds.

**Comment by cala** [comment]
I agree that creatures in Elwynn Forest repop quite fast, it is especially visible in caves. This is the case in many other places.

However, most of developers agree that blizz is real-time adjusting respawn time on their server depending on the number of players in an area. CMaNGOS core do not have this feature.

So, our only choice is to fix a middle ground value in DB. The question is: how can we determine a "right" value, one that (almost) everyone will agree on?

**Comment by cala** [comment]
I thought about that a good number of times (especially after being chain killed by easy NPCs that were repoping too fast in my back). I still don't have an acceptable solution.

**Comment by Schmoozerd** [comment]
Maybe apply a population respawn scaling in the core:

CONFIG_VARIABLE (default 0) <-- disable such scaling
scaling dependend on players in zone of level +-10 of npc-entry

```
RespawnDelay *= (1 + CONFIG  / (A * #PlayersInZoneAroundLevel + B))

with some nice values for A and B 
```

**Comment by ghost** [comment]
That's a very interesting idea. Does anyone know if the 120 second respawn timer is the "minimum" timer Blizzard uses?

**Comment by evil-at-wow** [comment]
It's certainly not a minimum possible timer these days. When an expansion launches, there are a lot of players in the new zones leveling, and the first week of the expansion most quest related mobs are respawning after seconds! I admit that is a rather special condition, but it shows they can scale from seconds to whatever value they use for "deserted" zones (I assume that those zones have the maximum value).

**Comment by Sinoxan** [comment]
An archived page of wowpedia mentions that the respawn time for quest related mobs was 60 seconds and that for regular mobs the respawn time was around 15 minutes, that seems quite high though.  They had a page specifically on respawn times however it was deleted and deleted pages can only be viewed by admins.

I did some searching and found a couple of pages that mentioned respawn times, just search for spawn and it should take you to the right place on the page.

Guide on farming linen cloth mentions mobs with 30 second respawn time
http://www.elitepvpers.com/forum/wow-exploits-hacks-tools-macros/38985-westfall-instant-respawn-mobs.html

~40 second respawn time for quest mob mentioned
http://wow-guide.blogspot.co.uk/2006/09/hunter-pets-guide.html

This mentions that high pop server would like a lower respawn time for world bosses, which probably have a specific respawn time rather than a variable.
http://forum.rpg.net/archive/index.php/t-287873.html

Article on farming fast spawn time mobs
http://wow.allakhazam.com/wiki/wow_guide_secret_hunting_places

Patch 1.11.0 mentions changing respawn time of individual mobs, making some spawn quicker, and making another mob stop respawning instantly.
http://us.battle.net/wow/en/game/patch-notes/1-11-0

This article from December 2004 gives the impression that the respawn times for mobs are fixed.
http://www.gamespot.com/articles/world-of-warcraft-walkthrough/1100-6114179/

Article talking about dynamic respawn rates introduced in TBC
http://tobolds.blogspot.co.uk/2007/01/dynamic-spawn-system.html

This also says that the dynamic respawn system wasn't introduced until 2.0
http://www.wowinterface.com/forums/showthread.php?t=11522

**Comment by PatSmuk** [comment]
That's really interesting. Thanks for doing the research.

I would imagine it's based on proximity to the dead creature rather than being in the same zone. But apparently it's a post-TBC thing. Could have it as an optional feature in classic though.

**Comment by cala** [comment]
I agree with @Patman64 thanks a lot for these researches.
From what I understand, the system proposed by @Schmoozerd should only be implemented for TBC core and beyond.
This means that for Classic either we find the correct static values for most of the creatures or we also implement the system: the various core version will be more compatible, but Classic will not be blizzlike on this specific point.

**Comment by alexluana** [comment]
it can be added with an option in mangos.conf for classic in this point...so for the default is deactivated, for rest of cores to be activated by default... but the idea is wonderful to have such system

**Comment by TheOnlyJoey** [comment]
Good day,

I am running into this problem as well. I have hosted a vanilla server for a little experiment and am running into problems where some area's and caves are impossible to clear due to the way to high respawn time.
I have compared with official server, and the respawn time there seem at least double or triple as long on low population servers.

I agree on the idea that a global value (modifier that can set the respawn time to 1.2x, 2.0x, 0,5x etc) might be useful for multiple cases. Of course this does not have to be enabled by default (like a lot of mangos.conf variables) but can help out quite a lot.
Currently our server is almost unplayable and will be since we have a max of 10 players on the server in different regions.

**Comment by bipbopbupbip** [comment]
I also noticed this issue when I was flying in the Ohn'ahran Plains (at that time, it was already TWW).
At the spot with neutral mobs, the respawn time was less than 10 seconds if you killed them more or less simultaneously.

[wowguide](
http://wow-guide.blogspot.com/2006/09/hunter-pets-guide.html)

and [wow](https://chaosboost.com/database/siesbarg-mount-world-of-warcraft/) chaosguide 

I also used this link shared by someone above because I wanted to find information about the spot I discovered on that site, but the link had been deleted. :(

All links like WOW Guide seem to have stopped working, so I asked Blizzard about it. After that, I got a response along the lines of: "Thank you very much for the information; we will pass it on to the developers."

lol
